### ISSUE TITLE
Nested Model in Union gives AttributeError: model_fields 

### ISSUE THREAD

**Issue by wynneevans** [RenaudLN/dash-pydantic-form#17]
Apologies if this is the wrong way to communicate but I am trying get an example working that matches the pydantic models I am trying to render in a dash app.

I have taken the example in your Discriminate Unions doc section and extended it by creating an Office class which is then used as the type for an additional attribute called 'options' in HomeOffice. I have done this only because I am stuck with multiple pydantic models that have this structure. I have noticed that if HomeOffice instead inherits Office then it works.

But is there any way of getting the pydantic model structure below to work with dash-pydantic-form ??

![image](https://github.com/RenaudLN/dash-pydantic-form/assets/25059495/35d45280-4352-429d-a85b-e8b2409e3eee)
![image](https://github.com/RenaudLN/dash-pydantic-form/assets/25059495/325535f8-aee3-4dc4-bd28-1c6aa3464cad)
![image](https://github.com/RenaudLN/dash-pydantic-form/assets/25059495/a244876c-59c8-432a-a9ad-ec9c37650556)

**Comment by RenaudLN** [comment]
Hey @wynneevans, thanks for bringing this to my attention. I am able to reproduce this issue and indeed there are issues with nested models in discriminated unions. I'll find a fix for this and let you know when there is progress on this!

**Comment by wynneevans** [comment]
That would be really amazing. Thank you for making this open source! If you do commit time to this issue then it would be even more amazing if it could handle multiple layers of nesting. i.e. In the above example, instead of Office inheriting BaseModel it inherited another class, which inherited another class, etc, until a class that inherits BaseModel. I've attached an example of the type of pydantic model I am trying to render in dash. This only in case you would like to stress test dash-pydantic-form against a real-world horrible structure. Changed to .txt file because .py files cannot be uploaded - it's not strictly working code just illustration of model structure.
[post.txt](https://github.com/user-attachments/files/16191013/post.txt)
